### PR TITLE
refactor args, use name as server id

### PR DESCRIPTION
**Pull request by themanforfree**

### CODE REVIEW

File: xline/src/main.rs [hunk 1]
```diff
@@ -162,6 +156,19 @@ struct ServerArgs {
...
+    for pair in s.split(',') {
+        if let Some((id, addr)) = pair.split_once('=') {
+            let _ignore = map.insert(id.to_owned(), addr.to_owned());
+        } else {
+            return Err("parse members error".to_owned());
+        }
+    }
+    Ok(map)
+}
+
```
Comment (rogercloud) [inline xline/src/main.rs:171]: Can we deduplicate the code with that in benchmark?

Comment (themanforfree) [reply xline/src/main.rs:171]: Add another utils crate? or let lock_utils become a more universal crate?

Comment (Phoenix500526) [reply xline/src/main.rs:171]: I prefer to let `lock_utils` become more universal. The name of `lock_utils` seems to be too concrete,  or we can rename it `utils`. The `lock_utils` is one of the `utils`.

**Review by Phoenix500526 — APPROVED** [review]

File: xline/src/server/xline_server.rs [hunk 2]
```diff
@@ -31,23 +33,14 @@ type CurpServer = Rpc<Command>;
 #[allow(dead_code)] // Remove this after feature is completed
 #[derive(Debug)]
 pub struct XlineServer {
-    /// Server name
-    name: String,
-    /// Address of peers
-    peers: Vec<SocketAddr>,
+    /// State of current node
+    state: Arc<RwLock<State>>,
```
Comment (rogercloud) [inline xline/src/server/xline_server.rs:37]: why is it protected in a rwlock? I can't find a place to write to it. All references are read-only accesses.

Comment (themanforfree) [reply xline/src/server/xline_server.rs:37]: when I can get new leader_id from curp, the state will be updated, but in this PR it is read-only

Comment (rogercloud) [reply xline/src/server/xline_server.rs:37]: OK, I see.

**Review by rogercloud — APPROVED** [review]

**Review by Phoenix500526 — APPROVED** [review]

**Pull request merged**